crates/turborepo-lib/src/run/scope/filter.rs

```diff
@@ -176,21 +177,34 @@ impl<'a, T: GitChangeDetector> FilterResolver<'a, T> {
                 .map(|(name, _)| name.to_owned())
                 .collect()
         } else {
-            self.get_packages_from_patterns(patterns)?
+            self.get_packages_from_patterns(affected, patterns)?
         };
 
         Ok((filter_patterns, is_all_packages))
     }
 
     fn get_packages_from_patterns(
         &self,
+        affected: &Option<(String, String)>,
         patterns: &[String],
     ) -> Result<HashSet<PackageName>, ResolutionError> {
-        let selectors = patterns
+        let mut selectors = patterns
             .iter()
             .map(|pattern| TargetSelector::from_str(pattern))
             .collect::<Result<Vec<_>, _>>()?;
 
+        if let Some((from_ref, to_ref)) = affected {
+            selectors.push(TargetSelector {
+                git_range: Some(GitRange {
+                    from_ref: from_ref.to_string(),
```

Comment:
I don't think this is correct, we need to be comparing the base of the branch and not just `main`. I could be wrong, but I think these are distinct. Can we verify this is correct with a test where the branch base isn't the tip of `main`?

Reply:
Will add a test, but I believe this is correct, because we do `git diff from...to`, which is equivalent to `git diff $(git merge-base from to) to`